apps/laboratory/src/components/Wagmi/WagmiSendCallsCustomAbiWithPaymasterServiceTest.tsx

```diff
@@ -0,0 +1,288 @@
+import { useCallback, useEffect, useMemo, useState } from 'react'
+
+import { Button, Input, Select, Stack, Text, Tooltip } from '@chakra-ui/react'
+import { type Abi, encodeFunctionData, parseEther } from 'viem'
+import { useAccount } from 'wagmi'
+import { useSendCalls } from 'wagmi/experimental'
+
+import { useAppKitAccount } from '@reown/appkit/react'
+
+import { useChakraToast } from '@/src/components/Toast'
+import { useWagmiAvailableCapabilities } from '@/src/hooks/useWagmiActiveCapabilities'
+import { EIP_5792_RPC_METHODS, WALLET_CAPABILITIES } from '@/src/utils/EIP5792Utils'
+
+function getWriteMethodsFromAbi(abi: string) {
+  try {
+    const abiJson: Abi = JSON.parse(abi)
+
+    const writeFunctions = abiJson.filter(
+      item =>
+        item.type === 'function' &&
+        (item.stateMutability === 'nonpayable' || item.stateMutability === 'payable')
+    )
+
+    return writeFunctions.filter(func => 'name' in func).map(func => func.name)
+  } catch (error) {
+    return []
+  }
+}
+
+export function WagmiSendCallsCustomAbiWithPaymasterServiceTest() {
+  const {
+    provider,
+    supported,
+    supportedChains: capabilitySupportedChains,
+    currentChainsInfo
+  } = useWagmiAvailableCapabilities({
+    capability: WALLET_CAPABILITIES.PAYMASTER_SERVICE,
+    method: EIP_5792_RPC_METHODS.WALLET_SEND_CALLS
+  })
```

Comment:
not for this PR but curious since you touched this, wyt about moving this to a higher level / caching somehow ? capabilities won't change during connection and we're doing 1 request per test

Reply:
I think this is better to keep it simplistic so people can copy to test their implementation